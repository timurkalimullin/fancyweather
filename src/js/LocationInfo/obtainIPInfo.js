export default async function obtainIPInfo(apiKey) {
  const url = `https://ipinfo.io/json?token=${apiKey}`;
  const res = await fetch(url);
  if (!res.ok) {
    throw new Error('IPinfoAPI network problems, can`t find you');
  }
  const data = await res.json();
  return data;
}
