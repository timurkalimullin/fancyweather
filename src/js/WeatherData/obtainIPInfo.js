export default async function obtainIPInfo(apiKey) {
  const url = `https://ipinfo.io/json?token=${apiKey}`;
  const res = await fetch(url);
  const data = await res.json();
  return data;
}

// city: "Saarbr√ºcken"
// country: "DE"
// hostname: "136.ip-54-37-73.eu"
// ip: "54.37.73.136"
// loc: "49.2333,7.0000"
// org: "AS16276 OVH SAS"
// postal: "66123"
// region: "Saarland"
// timezone: "Europe/Berlin"
