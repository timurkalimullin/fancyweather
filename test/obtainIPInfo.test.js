const fetchMock = require('fetch-mock');
import obtainIPInfo from '../src/js/LocationInfo/obtainIPInfo';

const apiKeyIPInfo = 'ad4cb296462880';

describe('obtainIPInfo', () => {
  beforeEach(() => {
    fetch.resetMocks()
  })
  it ('should return JSON file with property "loc"', async () => {
    fetch.mockResponseOnce(JSON.stringify({ loc: '345, 567' }));
    const response = await obtainIPInfo(apiKeyIPInfo);
    expect(response.hasOwnProperty('loc')).toMatchSnapshot();
  });
});

