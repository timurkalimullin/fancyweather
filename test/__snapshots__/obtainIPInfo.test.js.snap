// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`obtainIPInfo should return JSON file with property "loc" 1`] = `true`;
